modules/nixos.nix

```diff
@@ -165,5 +164,36 @@ in {
           concatStringsSep " " ruleString
       ) (filter (f: f.enable && f.source != null) (attrValues files));
     }) (filterAttrs (_: u: u.files != {}) config.homes);
+    systemd.services = mapAttrs' (name: {files, ...}: {
+      name = "hjem-monitor-" + name;
+      value.enable = true;
+      value.wants = ["systemd-tmpfiles-setup.service" "nix-daemon.socket"];
+      value.after = ["nix-daemon.socket"];
+      value.wantedBy = ["default.target"];
+      value.description = "Monitoring for Hjem files";
+      value.serviceConfig = {
+        Type = "exec";
+        ExecStart = let
+          serviceScript = pkgs.writeScript "hjem-monitor-script" ''
```

Comment:
If bash is used, prefer `writeShellApplication` here, so that we're not blowing our feet off. Actually I'd really like to not use bash, if possible.

Reply:
Why? What is the difference? also bash is ultimately the best choice here due to the simplicity of the checker.

Reply:
Not-using bash was just a musing that outside the PR scope, I should've made that clearer. and `writeShellAplication` runs shellcheck to prevent accidental issues, see https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-writeShellApplication

I forgot about `.script` in the systemd module, and in that cate, please set `enableStrictShellChecks` as well.